test/python/WMCore_t/MicroService_t/MicroService_t.py

```diff
@@ -80,28 +82,86 @@ def testGetStatus(self):
-
-        params = {"service": "transferor"}
-        data = self.mgr.getdata(url, params=params, encode=True, decode=True)
-        self.assertEqual(data['result'][0]['microservice'], self.managerName)
-        self.assertEqual(data['result'][0]['api'], api)
+        # perform check with and without gzip encoding
+        for hdr in ['', 'gzip']:
+            headers = {'Accept-Encoding': hdr}
+            data = self.mgr.getdata(url, params=params, headers=headers, encode=True, decode=True)
+            # check if our data is returned as bytes (gzip encoding)
+            if isinstance(data, bytes) and hdr == 'gzip':
+                data = gzip.decompress(data)
+                if isinstance(data, bytes):
+                    data = data.decode("utf-8")
+                data = json.loads(data)
+            print("### testGetStatus, headers=", headers, "data=", data, type(data))
+            # check that result key is present in data
+            if 'result' in data:
+                self.assertEqual(data['result'][0]['microservice'], self.managerName)
+                self.assertEqual(data['result'][0]['api'], api)
+            else:
+                print("### result is not present in data")
+                print(data)
+                self.assertEqual('result' in data, True)
+
+            params = {"service": "transferor"}
+            data = self.mgr.getdata(url, params=params, headers=headers, encode=True, decode=True)
+            # check if our data is returned as bytes (gzip encoding)
+            if isinstance(data, bytes) and hdr == 'gzip':
+                data = gzip.decompress(data)
+                if isinstance(data, bytes):
+                    data = data.decode("utf-8")
+                data = json.loads(data)
+            print("### testGetStatus, headers=", headers, "data=", data, type(data))
+            # check that result key is present in data
+            if 'result' in data:
+                self.assertEqual(data['result'][0]['microservice'], self.managerName)
+                self.assertEqual(data['result'][0]['api'], api)
+            else:
+                print("### result is not present in data")
+                print(data)
+                self.assertEqual('result' in data, True)
 
     def testGetInfo(self):
         "Test function for getting state of the MicroService"
         api = "status"
         url = '%s/%s' % (self.url, api)
         params = {}
-        data = self.mgr.getdata(url, params=params, encode=True, decode=True)
-        self.assertEqual(data['result'][0]['microservice'], self.managerName)
-        self.assertEqual(data['result'][0]['api'], api)
-
-        params = {"request": "fake_request_name"}
-        data = self.mgr.getdata(url, params=params, encode=True, decode=True)
-        self.assertEqual(data['result'][0]['microservice'], self.managerName)
-        self.assertEqual(data['result'][0]['api'], api)
+        # perform check with and without gzip encoding
```

Comment:
For these unit tests, it would be cleaner if you actually create new unit tests. For instance:
* `def testGetInfo` which will retrieve a standard json uncompressed data
* `def testGetInfoGzip` which will retrieve a compressed data.

It's easier to review and maintain.

Reply:
My first impression about your comment was  may be it is good idea, but my second thought is not since it will re-duplicate the entire code of tests. The for loop is sufficient to make both tests and the data should arrive and handled regardless of compression. I'll rather see how to improve the tests logic instead of duplicating entire tests function to do the same stuff to only retrieve data in gziped format.

Reply:
Unit tests are supposed to be small, and given that the setup for these tests is close to 0, separate methods with testing different functionalities is recommended. Better duplicating a couple of lines than relying on if/else here and there, IMO.

Reply:
ok, done, I separated the tests